src/types/mnemonic.ts

```diff
@@ -0,0 +1,23 @@
+import { Status } from './util';
+
+export type MnemonicOptions = {
+  contactRefId?: string;
+  thresholdMin?: number;
+  thresholdMax?: number;
+  deviation?: number;
+  precision?: number;
+};
+
+export type Mnemonic = {
+  id: string;
+  mnemonicId: string;
+  status: Status;
+  unit: string;
+  thresholdMax: number;
+  thresholdMin: number;
+  currentValue: number;
+  subsystem: string;
+  childSubSystem: string;
+  measurement: string;
+  contactRefId: string | undefined;
```

Comment:
I think we should make this type string only and just return an empty string if none provided so the property is always there. Ref comment on generate-mnemonic.ts file on line 26